web/src/app/util/PageActions.js

```diff
@@ -8,104 +8,90 @@ const PageActionsContext = React.createContext({
 })
 PageActionsContext.displayName = 'PageActionsContext'
 
-export class PageActionContainer extends React.PureComponent {
-  render() {
-    return (
-      <PageActionsContext.Consumer>
-        {({ actions }) => actions}
-      </PageActionsContext.Consumer>
-    )
-  }
+export const PageActionContainer = () => {
+  const { actions } = useContext(PageActionsContext)
+  return actions
 }
 
-export class PageActionProvider extends React.PureComponent {
-  state = {
-    actions: null,
-  }
-
-  _mountCount = 0
-  _mounted = false
-
-  componentDidMount() {
-    this._mounted = true
-    if (this._pending) {
-      this.setActions(this._pending)
-      this._pending = null
-    }
-  }
+export const PageActionProvider = (props) => {
+  const [actions, setActions] = useState(null)
 
-  componentWillUnmount() {
-    this._mounted = false
-    this._pending = false
-    this.setActions.cancel()
-  }
+  let _mountCount = 0
+  let _mounted = false
+  let _pending = null
```

Comment:
I think the way these are evaluated in classes vs. hooks is a bit different than it seems- `this.foo = 'bar'` is a synchronous update, and that value will live in the "background" of the component allowing it to retain its value on every re-render. With hooks though, these will get reset back to `0`, `false`, and `null` on every re-render.

I think we can get around this in hooks if we initialize the variables outside of the component and use the `useEffect` hook, as it will only update the values on mount and unmount:

```js
let _mountCount = 0
let _mounted = false

const MyComponentExample = () => {
  useEffect(() => {
    // componentDidMount code here

    return () => {
      // componentWillUnmount code here
    }
  }, [])

  // ...
}
```

Reply:
Having them outside of the component could introduce really odd side-effects bugs/leaks, we can use `useRef` though -- https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables

And then `useEffect(..., [])` to handle the mount/unmount triggers.

Reply:
Hey thanks! I'll take a look at it and make the changes

Reply:
Sure, I'll take a look at it.

Reply:
I've made some changes. please take a look and let me know if it looks good.